solvcon/parcel/gas/inout.py

```diff
@@ -65,12 +65,28 @@ def __init__(self, cse, show_bclist=False, perffn=None, **kw):
         super(MeshInfoHook, self).__init__(cse, **kw)
 
     def preloop(self):
+        from ...boundcond import BC
+        flag_generic_bc = False
         blk = self.blk
         self.info("Block information:\n  %s\n" % str(blk))
         if self.show_bclist:
             for bc in blk.bclist:
                 self.info("  %s\n" % bc)
 
+        # explicit info to prompt users to watch out issues like issue #177
+        for bc in blk.bclist:
+            if bc.__class__ is BC:
+                flag_generic_bc = True
+        if flag_generic_bc:
+            self.info("\n")
+            self.info("  One of the boundary conditions is generic boundary "
+                      "type.\n")
+            self.info("  Unexpected boundary conditions may be assigned.\n")
+            self.info("\n")
+            self.info("  Suggestion: Check your boundary condition mapper of "
+                      "your driving script.\n")
+            self.info("\n")
+
```

Comment:
By default, a generic class BC (boundary condition abstract class) will be used if there is no expected boundary type was found in the boundary condition mapper. However, the driving script of gas crashed because BC class has no attribute like soln:

```
AttributeError: ('[BC#0 "b\'wall\'": 1704 faces with 0 values]', 'soln', "'BC' object has no attribute 'soln'")
```

Type check so we could catch similar issue in the future. If all of the boundary condition type could be found from the bcoundary condition mapper, the driving script will show nothing. If there is any type is the abstract BC type, the driving script will throw out the prompt. It looks like:

```
......
*** ****************************************************************************
*** *** Start run_preloop ... 
*** *** ************************************************************************
*** *** Steps 0/600
*** *** Block information:
*** ***   [Block (3D/centroid): 1160 nodes, 8934 faces (1832 BC), 4009 cells]
*** *** 
*** ***   One of the boundary conditions is generic boundary type.
*** ***   Unexpected boundary conditions may be assigned.
*** *** 
*** ***   Suggestion: Check your boundary condition mapper of your driving script.
*** *** 
*** *** Traceback (most recent call last):
  File "./go", line 234, in <module>
    sc.go()
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/site-packages/solvcon/cmdutil.py", line 190, in go
    cmd()
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/site-packages/solvcon/command.py", line 1094, in __call__
    func(submit=False, **funckw)
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/site-packages/solvcon/case.py", line 1085, in simu
    cse.run(level=runlevel)
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/site-packages/solvcon/case.py", line 841, in run
    self._run_preloop()
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/site-packages/solvcon/case.py", line 876, in _run_preloop
    self.solver.solverobj.apply_bc()
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/site-packages/solvcon/parcel/gas/solver.py", line 149, in apply_bc
    self.call_non_interface_bc('soln')
  File "/home/tai271828/my-local-installation/miniconda3/solvcon-dev/lib/python3.5/site-packages/solvcon/solver.py", line 424, in call_non_interface_bc
    getattr(bc, name)(*args, **kw)
AttributeError: ('[BC#0 "b\'wall\'": 1704 faces with 0 values]', 'soln', "'BC' object has no attribute 'soln'")
```